code/_onclick/click.dm

```diff
@@ -129,7 +129,7 @@
 
 	// Allows you to click on a box's contents, if that box is on the ground, but no deeper than that
 	if(isturf(A) || isturf(A.loc) || (A.loc && isturf(A.loc.loc)))
-		if(A.Adjacent(src)) // see adjacent.dm
+		if(A.Adjacent(src) || W.reach >= get_dist(src, A)) // see adjacent.dm

```

Comment:
Well, I woke up thirty minutes ago, so I guess I'm just tired. Here's how I'm reading it:

Adjacent is true if they're nearby.
Reach is true if they're within a certain range

Say we have the prank gun, which works on only your tile. And you have a mob on your tile that you want to prank.

Both checks pass. Mob gets pranked. YouTube ad revenue goes through the roof.

Now the mob moves one tile above you.

The reach check fails, so we check Adjacent. That one succeeds, because the mob is indeed adjacent.

Thus, the hit succeeds, because it uses _either_ check and not _both_ checks.

Analogies aside, it's super easy to change this, and I likely will later today to support these kind of weapons.